src/auth/models/strategy.class.ts

```diff
@@ -256,14 +258,41 @@ export abstract class Strategy extends events.EventEmitter {
 
     public makeAuthorization = (passport: any) => `Bearer ${passport.user.tokenset.accessToken}`
 
-    public setHeaders = (req: Request, res: Response, next: NextFunction): void => {
+    public setHeaders = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
         if (req.session?.passport?.user) {
-            req.headers['user-roles'] = req.session.passport.user.userinfo.roles.join()
-            req.headers.Authorization = this.makeAuthorization(req.session.passport)
+            if (
+                this.options.routeCredential &&
+                this.options.routeCredential.routes &&
+                this.options.routeCredential.routes.includes(req.url)
+            ) {
+                let routeCredentialToken
+                if (req.session.routeCredentialToken) {
+                    routeCredentialToken = req.session.routeCredentialToken
+                } else {
+                    routeCredentialToken = await this.generateToken()
+                    req.session.routeCredentialToken = routeCredentialToken
+                }
+                if (routeCredentialToken && routeCredentialToken.access_token) {
+                    req.headers.Authorization = `Bearer ${routeCredentialToken.access_token}`
+                }
+            } else {
+                req.headers['user-roles'] = req.session.passport.user.userinfo.roles.join()
+                req.headers.Authorization = this.makeAuthorization(req.session.passport)
+            }
         }
         next()
     }
 
+    public generateToken = async (): Promise<any | undefined> => {
+        const url = this.getUrlFromOptions()
+        try {
+            const response = await http.post(url)
+            return response.data
+        } catch (error) {
+            this.logger.error('error => ', error)
```

Comment:
Can we have a more specific error here 'error generating authentication token '

Reply:
Agreed